pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularAboutHealthyPexTask.java

```diff
@@ -0,0 +1,232 @@
+
+                var change = questionsToUpdate.get(stableId);
+                assert change != null;
+
+                var expectedExpressionText = StringUtils.normalizeSpace(change.getString(Keys.ChangeContent.EXPECTED));
+                var updatedExpressionText = StringUtils.normalizeSpace(change.getString(Keys.ChangeContent.UPDATED));
+
+                // If there isn't a pre-existing expression, just bail. This patch
+                // doesn't have any cases where a new Expression is being created
+                var expression = expressionSql.getById(block.getShownExprId()).orElseThrow();
+                var expressionText = StringUtils.normalizeSpace(expression.getText());
+
+                if (Objects.equals(expressionText, updatedExpressionText)) {
+                    log.info("The shown expression for [{}] was already updated, skipping...", readableName);
+                    return;
+                }
+
+                if (!Objects.equals(expressionText, expectedExpressionText)) {
+                    var message = String.format("The shown expression for [%s] does not match the expected value.", readableName);
+                    log.error("{}\n\texpected: '{}'\n\tactual:   '{}'", message, expectedExpressionText, expressionText);
+                    throw new DDPException(message);
+                }
+
+                log.info("Found expression {} for [{}].", expression.getGuid(), readableName);
+
+                var updateExpression = new Expression(expression.getId(), expression.getGuid(), updatedExpressionText);
+                var result = expressionSql.update(updateExpression);
+                DBUtils.checkInsert(1, result);
+
+                log.info("Successfully updated expression {} for [{}].", expression.getGuid(), readableName);
+            });
+
+        daoCleanup();
+
+        log.info("TASK:: {} has completed successfully.", taskName);
+    }
+
+    private void daoSetup(Handle handle) {
+        this.studySql = handle.attach(JdbiUmbrellaStudy.class);
+        this.activitySql = handle.attach(JdbiActivity.class);
+        this.activityVersionSql = handle.attach(JdbiActivityVersion.class);
+        this.expressionSql = handle.attach(JdbiExpression.class);
+        this.activityDao = handle.attach(FormActivityDao.class);
+    }
+
+    private void daoCleanup() {
+        this.studySql = null;
+        this.activitySql = null;
+        this.activityVersionSql = null;
+        this.expressionSql = null;
+        this.activityDao = null;
+    }
+
+    /** 
+     * Finds the set of parent blocks in an activity for a given set of question stable ids.
+     * 
+     * <p>This will only return blocks for which the question is an immediate child. If
+     * the question is embedded in any other structure, it will not be found
+     * 
+     */
+    private List<QuestionBlockDef> findBlocksForQuestions(FormActivityDef activityDefinition, Set<String> stableIds) {
```

Comment:
Wondered if this can be added some Dao so that it can be reused. May be it can be part of bigger scope where we can try to come up with some generic patches which can update study config (patches like UpdateTemplatesInPlace)

Reply:
That piece of code is highly tailored to this specific situation. If there's a need for something like that, this may be useful inspiration, but I would not consider this safe for general use, and the performance characteristics if it would need a careful evaluation.